### PR TITLE
Chore/prettier config

### PR DESCRIPTION
**Pull request by alxtz**
# Feature
* move prettier check/write hook to pre-commit position, right now the code would be auto-formatted & added when you do a commit.

* every time you type `git commit`, the git hooks will autorun `prettier --write` and `git add` for you on those files.

# Notes
* Might have to run yarn again to setup those git hooks ( `husky` uses npm's post-install hooks the setup those things )

* Can see in `xfers-sdk/.git/hooks/pre-commit` to see if it's set up properly

* Setting up prettier in precommit hook is the prefered way in prettier's doc https://prettier.io/docs/en/precommit.html

### CODE REVIEW

**Comment by henghonglee** [comment]
coolios, but still we should make people install prettier on their IDE, how can we tell if they do/do-not?

**Comment by alxtz** [comment]
> coolios, but still we should make people install prettier on their IDE, how can we tell if they do/do-not?

Yeah, originally we use the `"pre-push": "yarn prettier-check"` gate to ensure every developer have been using `RuboCop` to do the *format-when-we-save-file* workflow

This PR is a design / flavor level of change on this workflow, which could make setting prettier up on IDE optional, and all the checking + formatting would just happens automatically when typing git commands, not to couple with individual’s IDE setup

Motivations:
1. trying to make code auto-formatting in IDE less distracting
2. prefer "auto fixing" over "breaking current workflow" for better DX
3. makes it less troublesome to tell every to set up their IDE, those git hooks are automatically set up by husky after every `npm install` using npm's `pre-install script` https://stackoverflow.com/questions/46556921/npm-preinstall-script

Anyway, this is just workflow change for the SDK (have done discussed), feel free to discuss in case of consistency

File: JavaScript/package.json [hunk 1]
```diff
@@ -10,9 +10,7 @@
     "test:watch": "jest --watch",
     "esmbuild": "rollup --config",
     "sg:nightwatch": "nightwatch --tag singapore --retries 3",
-    "id:nightwatch": "nightwatch --tag indonesia --retries 3",
-    "prettier-check": "prettier --check '**/*.js'",
-    "prettier": "prettier --write '**/*.js'"
```
Comment (henghonglee) [inline JavaScript/package.json]: you probably can keep this in the case people want to do `npm prettier`

Comment (alxtz) [reply JavaScript/package.json]: image
Done!

https://github.com/Xfers/xfers-sdk/pull/281/commits/ab42858fa54e5af8422023077c1c04e072f78ca9

**Pull request merged**